File: sandbox/exercism/ruby/2024/difference_of_squares.rb
class Squares
  def initialize(n)
    @to = n
  end

  def square_of_sum
    (1..@to).sum ** 2
  end

  def sum_of_squares
    (1..@to).sum { |n| n ** 2 }
  end

  def difference
    square_of_sum - sum_of_squares
  end
end

File: sandbox/exercism/go/2025/difference_of_squares.go
// https://exercism.org/tracks/go/exercises/difference-of-squares

package diffsquares

import "math"

func SquareOfSum(n int) int {
	sum := 0

	for i := 1; i <= n; i++ {
		sum += i
	}

	return int(math.Pow(float64(sum), 2.0))
}

func SumOfSquares(n int) int {
	result := 0.0

	for i := 1; i <= n; i++ {
		result += math.Pow(float64(i), 2.0)
	}

	return int(result)
}

func Difference(n int) int {
	return SquareOfSum(n) - SumOfSquares(n)
}
